### DISCUSSION TITLE
Generated code missing parameters for Upsert

### DISCUSSION THREAD

**Discussion by jamesleeht** [aarondl/sqlboiler#1196] [Q&A]
In the upsert documentation, I see this example:

```
// INSERT INTO pilots ("id", "name") VALUES ($1, $2)
// ON CONFLICT ("id") DO UPDATE SET "name" = EXCLUDED."name"
err := p1.Upsert(ctx, db, true, []string{"id"}, boil.Whitelist("name"), boil.Infer())
```

I'm a little confused which parameter this `[]string{"id"}` is supposed to go, since my generated code only has these parameters:

```
 Upsert(exec boil.Executor, updateColumns, insertColumns boil.Columns) error
```

Am I doing something wrong with the generation?

**Comment by stephenafamo** [comment]
From a first look, I assume you generated code for MySQL whose UPSERT syntax is different from Postgres/SQLite.

MySQL uses "ON DUPLICATE KEY UPDATE" and just looks for conflicts on the primary/unique keys. [Reference](https://dev.mysql.com/doc/refman/8.0/en/insert-on-duplicate.html).
